### ISSUE TITLE
New popups

### ISSUE THREAD

**Issue by sghpjuikit** [sghpjuikit/player#177]
Current popups have several UX issues and also have complex implementation (for example inheriting owner window's styling or problematic show/hide animation logic). I'm proposing new popup implementation.

.............................

Needs to solve issues:
- [x] problem with focus for popup with no window owner (cant receive key or mouse events)  
  Nasty issue that apparently affects any window with UTILITY owner (the only way to hide taskbar icon). This is because the window believes it is focused, yet gets no events and no amount of focusing works. Had to use `Robot()` and make an actual mouse click to bypass this.
- [x] problem with popup being on top of all system windows even if owner is in bgr  
  JavaFX issue. Using simple trick with `Stage.focus/alwaysOnTop` makes the popup behave as expected - on top of its owner, but not other windows
- [x] problem with popup not being resizable by system (affects PopOver.class)  
  Resizing/moving window natively by system/external program is now respected by the application.
- [x] problem with initial popup position when auto-sizing to content  
  This infuriating issue is gone. Notifications now display beautifully.
- [x] problem with FileChooser appearing behind popup/closing the popup due to focus loss
- [x] complicated/confusing/broken PopOver API
- [x] reflection dementiage in PopOver, Window inheritance 
- [x] auto-hide respecting popup hierarchy  
  Unfocused child popups close.
- [x] escape-hide respecting popup hierarchy  
  Focused popup gets closed.
- [x] popup passing events to owner instead of handling them  
  No more TAB/typing issues in popups
- [x] window focus pseudoclass  
  Added focus effect
- [x] make all ui styleable  
  Styles in `em`s, gap and paddings are all defined css. 

Needs to retain
- [x] styling  
  Simplified, no cssBridge magic, all styles are applied correctly. Styleclass set on root instead of window. Notification/helpPopup can easily override style.
- [x] animating  
  Simplified due to removal of `hideStrong()`. Added zoom in/out effect.
- [ ] moving with owner
- [x] smart positioning  
  Simplified. Uses `sizeToScene()` and deterministic show-resize-relocate semantics. Created dead simple API with single entry `show()` method
- [x] unfocusable popups  
  Simplified mess with window owner.

Can abandon
- [x] arrow pointer
- [x] some of the inherited Popup API

Can remove
- [x] PopOver & PopOverSkin & PopOverUtil  
  1700 lines of hacky Java code (changed to 500 lines of hack-free Kotlin code), 8 TODOs magically fixed

Known issues:
- `CTRL+W`/`CTRL+Q` closes application window instead of a popup only. Feature or bug?

**Comment by sghpjuikit** [comment]
I'm working on a popup based on Stage - normal windows. Most functionality is already working and the mentioned issues are also gone.

**Comment by sghpjuikit** [comment]
As can be seen, the problems were numerous and not exactly easy to fix, but I managed to fix all of them.

Last complication: unfocusable and focusable popups can not use the same approach to window owners - focusable need owner per popup and call `show()` and `focus`, unfocusable need one pre-made owner. No amount of magic would make these two cases work with same owner approach. This is problematic as the owner must be constructed in the `show()` method, which indirectly forces the popup to also be constructed in the `show()` method, which makes it unsafe for use and unsafe to be exposed in the API, which required some creativity to get around. It also makes it more difficult to reuse the window.

Merely figuring out these facts took me about 8 hours...

**Comment by sghpjuikit** [comment]
In the end I had to resort to using Popup windows for unfocusable windows. I was thinking about splitting these off to completely different class, but I have decided not to, for now. This only affects Notification, which is ok this way.

Another issue - File chooser closing windows with `autoHide=true` is an unsolvable issue (without workarounds), because JavaFX is unaware of the child window created and focused, hence can not defend against the focus loss.
I'm going to try solving this with Window.properties - a property that would request window to override `autoHide`. This will require all FileChoosers to use this property while they are open, but because we are using file chooser through an utility method, this should be easy.

**Comment by sghpjuikit** [comment]
```
var Window.hasFileChooserOpen: Boolean
   set(value) {
      if (value) {
         properties["no-auto-hide"] = Any()
      } else {
         properties -= "no-auto-hide"
         requestFocus()
      }
   }
   get() = "no-auto-hide" in properties
```

**Comment by sghpjuikit** [comment]
I'll implement `moveWithOwner` independently as I plan to make this feature more configurable that it was before.

**Comment by sghpjuikit** [comment]
Fixed in 53da6a4

**Comment by sghpjuikit** [comment]
This seems to have partially fixed the javafx focused css style application on text fields in settings when focus transitions to the group tree.
